test/sequential/test-cluster-disconnect-leak.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+// Flags: --expose-internals
+
+const common = require('../common');
+const assert = require('assert');
+const net = require('net');
+const cluster = require('cluster');
+const handles = require('internal/cluster').handles;
+const os = require('os');
+
+if (common.isWindows) {
+  console.log('1..0 # Skipped: This test does not apply to Windows.');
+  return;
+}
+
+cluster.schedulingPolicy = cluster.SCHED_NONE;
+
+if (cluster.isMaster) {
+  const worker1 = cluster.fork();
+  worker1.on('message', common.mustCall(() => {
+    worker1.disconnect();
+    for (let i = 0; i < tries; ++ i) {
+      const w = cluster.fork();
+      w.on('online', common.mustCall(w.disconnect));
+    }
+  }));
+
+  const cpus = os.cpus().length;
+  const tries = cpus * 16;
```

Comment:
Nit: Maybe move these declarations up above the `worker1.on('message', ...)` because someone reading the test will probably expect `tries` to be defined before it is used on line 22?